Sources/Xcodeproj/XcodeProjectModelSerialization.swift

```diff
@@ -410,10 +397,76 @@ extension PropertyListDictionaryConvertible {
         }
         return .dictionary(dict)
     }
+
+    /// Returns a property list representation of the build settings, in which
+    /// every struct field is represented as a dictionary entry.  Fields of
+    /// type `String` are represented as `PropertyList.string` values; fields
+    /// of type `[String]` are represented as `PropertyList.array` values with
+    /// `PropertyList.string` values as the array elements.  The property list
+    /// dictionary only contains entries for struct fields that aren't nil.
+    ///
+    /// Note: BuildSettings is a value type and PropertyListSerializable only
+    /// applies to classes.  Creating a property list representation is totally
+    /// independent of that serialization infrastructure (though it might well
+    /// be invoked during of serialization of actual model objects).
+    public func asPropertyList() -> PropertyList {
+        return type(of: self).asPropertyList(self)
+    }
 }
 
 extension Xcode.BuildFile.Settings: PropertyListDictionaryConvertible {}
-extension Xcode.BuildSettingsTable.BuildSettings: PropertyListDictionaryConvertible {}
+extension Xcode.BuildSettingsTable.BuildSettings: PropertyListDictionaryConvertible {
+    public func asPropertyList() -> PropertyList {
+        var buildSettings = self
+
+        // Space-separated setting is a setting whose value is split into multiple
+        // values using space as a separator.
+        //
+        // Example: ["value1", "value2 value3"] -> ["value1", "value2", "value3"]
+        // https://github.com/apple/swift-package-manager/pull/2770#issuecomment-638453861
+        let spaceSeparatedSettingKeyPaths: [WritableKeyPath<Xcode.BuildSettingsTable.BuildSettings, [String]?>] = [
+            \.FRAMEWORK_SEARCH_PATHS,
+            \.GCC_PREPROCESSOR_DEFINITIONS,
+            \.HEADER_SEARCH_PATHS,
+            \.LD_RUNPATH_SEARCH_PATHS,
```

Comment:
I don't think this one actually should be in this list, but the others are correct.

Reply:
From [Build Setting Reference](https://developer.apple.com/library/archive/documentation/DeveloperTools/Reference/XcodeBuildSettingRef/1-Build_Setting_Reference/build_setting_ref.html):

> Space-separated list of directory paths. Specifies the run-path locations at which the dynamic loader searches for the product’s run-path dependent libraries. See Dynamic Library Programming Topics.

Reply:
Did I address your concern regarding `LD_RUNPATH_SEARCH_PATHS` or it still needs to be removed?

Reply:
Hi, sorry for taking a while to respond.

So, there are two separate levels of quoting at work here.  The first is in string vs string-list build settings, and as you note, there is a list of them in the build settings reference.  This is part of the public interface of how build settings work.

The second aspect is in serialization, which is a quirk of Xcode's implementation.  There is a fixed set of settings that Xcode splits into arrays when archiving, and `LD_RUNPATH_SEARCH_PATHS` is not among them.  But it is able to read any of the settings as an array if it's written that way in the archive.  So it's fine to write it as an array instead of a string, and you're right to quote it.

Thanks for the fix!